hw11/Lesson11.py

```diff
@@ -0,0 +1,64 @@
+import datetime
+import inspect
+
+
+def my_decorator_func(func):
+    def deco_func(*args, **kwargs):
+        func_name = func.__name__
+        res = func(*args, **kwargs)
+        now_ = datetime.datetime.now()
+        print(f'The function name is {func_name} - was called at {now_.strftime("%H:%M")}\n')
+        return res
+    return deco_func
+
+
+@my_decorator_func
+def my_function():
+    print('This is my func')
+
+
+my_function()
+
+
+class ContextManager:
+    def __init__(self):
+        print('=' * 10)
+        print('What is the sense of this task?')
+
+    def __enter__(self):
+        # print('='*10)
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        print('=' * 10)
+        if exc_type:
+            print(exc_value)
+            return True
+
+
+with ContextManager() as manager:
+    print('This is the test Context manager without sense')
+
+print('\n')
+try:
+    print('='*10)
+    print('What is the sense of this task?')
```

Comment:
The sense of this taks is to show that we can do the same with different constructions in Python. What to use is up to you, but now you know how to control the execution of your code, handle error without raising them and have some parts that will be executed in any case